framework/source/class/qx/core/Property.js

```diff
@@ -891,6 +894,48 @@ qx.Bootstrap.define("qx.core.Property",
 
 
     /**
+     * Emit code for the equality check evaluation
+     *
+     * @param code {String[]} String array to append the code to
+     * @param clazz {Class} the class which originally defined the property
+     * @param config {Object} The property configuration map
+     * @param name {String} name of the property
+     */
+    __emitComparatorFunction : function (code, clazz, config, name)
+    {
+      code.push('var equ=');
+
+      if (typeof config.comparator === "function")
+      {
+        code.push('function(a,b){return !!', clazz.classname, '.$$properties.',
+                  name, '.comparator.call(this,a,b);};');
+      }
+      else if (typeof config.comparator === "string")
+      {
+        var members = clazz.prototype;
+        // Name of member?
+        if (members[config.comparator]!==undefined)
```

Comment:
So in addition to the first idea the string type comparator does not only contain some simple code which is wrapped in a function, but may also contain a reference to a class method by its name.
I think we should ommit this part and only have the inline code,
If we we call the inline code wrapped function in the context of the instance, we have access to all instance methods via ```this.```